package/MDAnalysis/coordinates/PDB.py

```diff
@@ -600,23 +608,24 @@ def _read_frame(self, frame):
             for line in f:
                 if line[:6] == 'ENDMDL':
                     break
-                # NOTE - CRYST1 line won't be found if it comes before the MODEL
-                # line, which is sometimes the case, e.g. output from gromacs
-                # trjconv
+                # NOTE - CRYST1 line won't be found if it comes before the
+                # MODEL line, which is sometimes the case, e.g. output from
+                # gromacs trjconv
                 elif line[:6] == 'CRYST1':
-                    A, B, C = map(float, [line[6:15], line[15:24], line[24:33]])
-                    alpha, beta, gamma = map(float, [line[33:40], line[40:47],
+                    A, B, C = np.float32([line[6:15], line[15:24],
+                                          line[24:33]])
+                    alpha, beta, gamma = np.float32([line[33:40], line[40:47],
                                                      line[47:54]])
                     self.ts._unitcell[:] = A, B, C, alpha, beta, gamma
                     continue
                 elif line[:6] in ('ATOM  ', 'HETATM'):
                     # we only care about coordinates
-                    self.ts._pos[pos] = list(map(float, [line[30:38],
-                                                         line[38:46],
-                                                         line[46:54]]))
+                    self.ts._pos[pos] = np.float32([line[30:38],
```

Comment:
I think it does. But the type casting is actually not needed as any value you put in an array is first type casted by the array. For instance:

```python
In [1]: import numpy as np
In [2]: a = np.zeros((4, 3))
In [3]: a[0, :] = '2.3 4.5 1.9'.split()  # The values we insert are strings
In [4]: a
Out[4]:
array([[ 2.3,  4.5,  1.9],
       [ 0. ,  0. ,  0. ],
       [ 0. ,  0. ,  0. ],
       [ 0. ,  0. ,  0. ]])
In [5]: a.dtype
Out[5]: dtype('float64')
In [6]: type(a[0, 0])
Out[6]: numpy.float64
```